src/main/java/com/aws/greengrass/device/ClientDevicesAuthService.java

```diff
@@ -99,9 +97,13 @@ public ClientDevicesAuthService(Topics topics, GroupManager groupManager, Certif
     protected void install() throws InterruptedException {
         super.install();
         //handleConfiguration
-        this.deviceGroupsTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
-        this.deviceGroupsTopics.subscribe(this::handleConfigurationChange);
-        this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC).subscribe(this::updateCAType);
+        this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((whatHappened, node) -> {
+            handleConfigurationChange(
+                    whatHappened, this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS)
+            );
+            updateCAType(whatHappened, this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC));
```

Comment:
Looks like the above two methods will be called on every config change, rather than just on device groups or ca type. To keep functionality similar to before, we'd need to check if the desired node changed before applying the update

Reply:
Addressed. Thanks!